plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/StandardDeployJob.java

```diff
@@ -68,37 +71,58 @@
   private IStatus cloudSdkProcessStatus = Status.OK_STATUS;
   private Process process;
 
-  private final StandardDeployJobConfig config;
+  private IProject project;
+  private Credential credential;
+  protected IPath workDirectoryParent;
+  private ProcessOutputLineListener stdoutLineListener;
+  private ProcessOutputLineListener stderrLineListener;
+  private DefaultDeployConfiguration deployConfiguration;
   private OutputCollectorOutputLineListener errorCollectingLineListener;
 
-  public StandardDeployJob(StandardDeployJobConfig config) {
+  public StandardDeployJob(IProject project,
+                           Credential credential,
+                           IPath workDirectoryParent,
+                           ProcessOutputLineListener stdoutLineListener,
+                           ProcessOutputLineListener stderrLineListener,
+                           DefaultDeployConfiguration deployConfiguration) {
     super(Messages.getString("deploy.standard.runnable.name")); //$NON-NLS-1$
-    this.config = Preconditions.checkNotNull(config, "config is null");
+    this.project = project;
+    this.credential = credential;
+    this.workDirectoryParent = workDirectoryParent;
+    this.stdoutLineListener = stdoutLineListener;
+    this.stderrLineListener = stderrLineListener;
+    this.deployConfiguration = deployConfiguration;
     errorCollectingLineListener =
-        new OutputCollectorOutputLineListener(config.getStderrLineListener(),
-                                              ERROR_MESSAGE_PREFIX);
+        new OutputCollectorOutputLineListener(this.stderrLineListener,
```

Comment:
I misread: I thought the wrappedListener had been brought inside the `if`.  Sorry!